k-distribution/include/builtin/domains.k

```diff
@@ -590,6 +590,98 @@ module STRING-BUFFER
   imports STRING-BUFFER-SYMBOLIC
 endmodule
 
+module BYTES-SYNTAX
+  syntax Bytes [hook(BYTES.Bytes)]
+
+  syntax Bytes ::= ".Bytes" [function, hook(BYTES.empty), impure]
+  syntax Endianness ::= "LE" [klabel(littleEndianBytes)]
+                      | "BE" [klabel(bigEndianBytes)]
+  syntax Signedness ::= "Signed" [klabel(signedBytes)]
+                      | "Unsigned" [klabel(unsignedBytes)]
+  syntax Int ::= Bytes2Int(Bytes, Endianness, Signedness) [function, hook(BYTES.bytes2int)]
+  syntax Bytes ::= Int2Bytes(Int, Int, Endianness) [function, hook(BYTES.int2bytes)]
+                 | Int2Bytes(Int, Endianness, Signedness) [function, klabel(Int2BytesNoLen)]
+  syntax String ::= Bytes2String(Bytes) [function, hook(BYTES.bytes2string)]
+  syntax Bytes ::= String2Bytes(String) [function, hook(BYTES.string2bytes)]
+
+  syntax Bytes ::= substrBytes(Bytes, Int, Int) [function, hook(BYTES.substr)]
+  syntax Bytes ::= replaceAtBytes(Bytes, Int, Bytes) [function, hook(BYTES.replaceAt)]
+  syntax Bytes ::= padRightBytes(Bytes, Int, Int) [function, hook(BYTES.padRight)]
+                 | padLeftBytes(Bytes, Int, Int) [function, hook(BYTES.padLeft)]
+  syntax Bytes ::= reverseBytes(Bytes) [function, hook(BYTES.reverse)]
+
+  syntax Int ::= lengthBytes(Bytes) [function, hook(BYTES.length), smtlib(lengthBytes)]
+endmodule
+
+module BYTES-SYMBOLIC [symbolic]
+  imports BYTES-SYNTAX
+  imports INT
+  imports K-EQUAL
+  imports STRING
+  imports STRING-BUFFER
+
+  syntax Bytes ::= "nilBytes"
+                 | Int ":" Bytes
+
+  rule .Bytes => nilBytes
```

Comment:
Why do we have two different constants for the empty byte-string? (`nilBytes` and `.Bytes`)

Reply:
.Bytes is a function, nilBytes is the constructor that the function evaluates to. This is done so that we don't have to change the attributes of the production based on the backend, which we can't do currently.